sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BridgeInternal.java

```diff
@@ -653,7 +654,8 @@ public static ConsistencyLevel fromServiceSerializedFormat(String consistencyLev
 
     @Warning(value = INTERNAL_USE_ONLY_WARNING)
     public static CosmosException createServiceUnavailableException(Exception innerException) {
```

Comment:
this method is being called by GoneAndRetryPolicy -> without passing in the substatuscode, the correct substatus code will still be getting lost.

Reply:
Fixed the code.